src/swarm/neo/connection/RequestOnConnBase.d

```diff
@@ -507,6 +614,185 @@ abstract class RequestOnConnBase
 
         /***********************************************************************
 
+            Waits for one of the specified events to occur and returns a
+            smart-union denoting which occurred.
+
+            There are four possible types of event:
+                1. Receiving a payload for this request.
+                2. Sending a payload.
+                3. The fiber being resumed after it was yielded and an epoll
+                   event-loop cycle occurred.
+                4. The fiber being resumed with a non-negative code.
+
+            The fiber is usually suspended while waiting for an event to occur.
+            The one exception is when sending -- the send may succeed
+            immediately, without needing to suspend the fiber.
+
+            Params:
+                flags = flags indicating whether to wait for receiving &/
+                    being resumed after yielding
+                fill_payload = optional delegate to fill in a payload to send.
+                    If this argument is null, sending does not occur
+
+            Returns:
+                an EventNotification instance denoting the event which occurred
+
+            Throws:
+                Exception on protocol or I/O error.
+
+        ***********************************************************************/
+
+        public EventNotification nextEvent (
+            Flags flags, FillPayloadDg fill_payload = null )
+        out ( fired_event )
+        {
+            auto fired_event_non_const = cast(EventNotification)fired_event;
+            assert(fired_event_non_const.active !=
+                fired_event_non_const.active.none);
+        }
+        body
+        {
+            EventNotification fired_event;
+
+            bool sending = fill_payload !is null;
+            bool receiving = (flags & Flags.Receive) > 0;
+            bool yielding = (flags & Flags.Yield) > 0;
+
+            // Set up sending, if required.
+            scope payload = this.new Payload;
```

Comment:
No, it can't since it would then be destroyed when the fiber is suspended here: https://github.com/sociomantic-tsunami/swarm/pull/44/files/3f9a330668b2e470541658a9f31d8d278d851d39#diff-58c716f1822ca1c61030b6c949624228R734

If I understand correctly.